### PR TITLE
gqa added to mpt attn

### PR DESCRIPTION
**Pull request by megha95**
mpt.py modified to support GQA in MPTAttention

### CODE REVIEW

**Comment by WoosukKwon** [comment]
@megha95 Thanks for submitting the PR! Just a quick question: Which MPT model uses GQA?

**Comment by megha95** [comment]
@WoosukKwon some of our customers want to try serving MPT with GQA. And, so this PR will help us enable it.

File: vllm/model_executor/layers/attention.py [hunk 1]
```diff
@@ -141,6 +141,8 @@ def forward(
                 else:
                     input_metadata.attn_bias = _make_alibi_bias(
                         self.alibi_slopes, batch_size, seq_len, query.dtype)
+                if self.num_kv_heads != self.num_heads:
+                    input_metadata.attn_bias._bias = input_metadata.attn_bias._bias.unflatten(1, (self.num_kv_heads, self.num_queries_per_kv))
```
Comment (WoosukKwon) [inline vllm/model_executor/layers/attention.py:145]: @megha95 Could you elaborate more on this case? I think the current RoPE-based GQA models do not need this. Is this for GQA + ALiBi?

**Review by WoosukKwon — APPROVED** [review]
LGTM. I fixed `attention.py` so that the change does not affect non-ALiBi models. Thanks for submitting the PR!

**Pull request merged**